sqld/src/replication/replica/replicator.rs

```diff
@@ -204,12 +203,20 @@ impl Replicator {
 
     async fn inject_frame(&mut self, frame: Frame) -> anyhow::Result<()> {
         let injector = self.injector.clone();
-        if let Some(commit_fno) =
-            spawn_blocking(move || injector.lock().inject_frame(frame)).await??
+        match spawn_blocking(move || injector.lock().inject_frame(frame)).await?
         {
-            self.meta.set_commit_frame_no(commit_fno).await?;
-            self.current_frame_no_notifier
-                .send_replace(Some(commit_fno));
+            Ok(Some(commit_fno)) => {
+                self.meta.set_commit_frame_no(commit_fno).await?;
+                self.current_frame_no_notifier
+                    .send_replace(Some(commit_fno));
+            }
+            Ok(None) => (),
+            Err(e @ crate::Error::FatalReplicationError) => {
+                // we conservatively nuke the replica and start replicating from scractch
```

Comment:
let's print the context of why we decided to self-nuke with `tracing::error!` level